librz/analysis/analysis.c

```diff
@@ -143,6 +143,7 @@ RZ_API RzAnalysis *rz_analysis_new(void) {
 			rz_analysis_add(analysis, analysis_static_plugins[i]);
 		}
 	}
+	analysis->ht_global_var = ht_pp_new0();
```

Comment:
this should be `ht_pp_new(NULL, global_kv_free, NULL);`.

And `global_kv_free` should probably free the `kv->key` and also use `rz_analysis_var_global_free` on the `kv->value`. That way whenever you delete an item from the Ht, it will also automatically free the global variable.